src/test/java/de/tum/in/www1/artemis/authorization/AuthorizationTestService.java

```diff
@@ -0,0 +1,100 @@
+    }
+
+    /**
+     * Returns all annotations on the given method that are relevant to authorization
+     *
+     * @param method the method to check
+     * @return List of relevant annotations
+     */
+    public List<Annotation> getAuthAnnotations(Method method) {
+        var annotations = Arrays.asList(method.getAnnotations());
+        return annotations.stream().filter(annotation -> ALLOWED_AUTH_METHOD_ANNOTATIONS.contains(annotation.annotationType())).toList();
+    }
+
+    /**
+     * Returns all annotations on the given class that are relevant to authorization.
+     *
+     * @param clazz the class to check
+     * @return List of relevant annotations
+     */
+    public List<Annotation> getAuthAnnotations(Class<?> clazz) {
+        var annotations = Arrays.asList(clazz.getAnnotations());
+        return annotations.stream().filter(annotation -> annotation.annotationType().equals(PreAuthorize.class)).toList();
+    }
+
+    /**
+     * Access a given annotation object and return the value of the annotation.
+     *
+     * @param annotation the annotation to access
+     * @return The value of the annotation
+     * @throws InvocationTargetException if the annotation does not have a value method
+     */
+    public String getValueOfAnnotation(Annotation annotation) throws InvocationTargetException, IllegalAccessException {
+        return (String) preAuthorizeValueAnnotation.invoke(annotation);
+    }
+
+    /**
+     * Wrapper method to add a new report to the report map
+     * Adds the report to the matching existing list, otherwise creates a new modifiable list
+     *
+     * @param reportsMap   The current collection of reports
+     * @param reportObject The report object the report is about
+     * @param report       The report to add
+     */
+    public <T> void addElement(Map<T, List<String>> reportsMap, T reportObject, String report) {
+        if (reportsMap.containsKey(reportObject)) {
+            reportsMap.get(reportObject).add(report);
+        }
+        else {
+            reportsMap.put(reportObject, new ArrayList<>(List.of(report)));
+        }
+    }
+
+    /**
+     * Prints the given reports to the console
+     * @param reportsMap The reports to print
+     */
+    public void printReports(Map<Class<?>, List<String>> reportsMap) {
+        System.err.println("Illegal configurations in classes:");
+        reportsMap.forEach((clazz, reportList) -> {
+            if (reportList.size() == 1) {
+                System.out.println("Class " + clazz.getSimpleName() + " has the following illegal configuration:");
+                System.out.println(reportList.get(0));
+            }
+            else {
+                System.out.println("Class " + clazz.getSimpleName() + " has the following illegal configurations:");
+                reportList.forEach(report -> {
+                    System.out.print(" - ");
+                    System.out.println(report);
+                });
+            }
```

Comment:
```suggestion
            String suffix = reportsList.size() > 1 ? "s" : "";
            System.out.printf("Class %s has the following illegal configuration%s:\n", clazz.getSimpleName(), suffix);
            reportList.forEach(report -> {
                System.out.print(" - ");
                System.out.println(report);
            });
```
Suggestion for duplication avoidance.

---

I personally also think that for a developer-only facing error message the optional plural-s can just always be there, as it does not really matter. Then the `suffix` could be removed and the `printf` reverted to the `println` as before.

Reply:
I changed this functionalty